### ISSUE TITLE
Support importing AMP from an npm package

### ISSUE THREAD

**Issue by Enriqe** [ampproject/amphtml#32110]
Currently importing from AMP from an npm package is not supported, since the file is not pre-processed before publication. 

See example PR https://github.com/ampproject/amphtml/pull/32076 and related https://github.com/ampproject/amphtml/pull/32109#discussion_r562182768

/cc @ampproject/wg-infra

**Comment by rsimha** [comment]
@Enriqe can you elaborate on what exactly needs to change here? I see from https://github.com/ampproject/amphtml/pull/32109#discussion_r562182768 that `extensions/amp-viewer-integration` now takes a `cdnProxyRegex` arg. What are you looking for from an infrastructure point of view to support this?

**Comment by Enriqe** [comment]
Yep, that's a workaround from the root issue, which is that you currently cannot import any AMP files from an npm package (i.e. [you can't do this](https://github.com/ampproject/amphtml/pull/32076/commits/6b8ebb139f0d3d50c35200e7222bfa02891f1d1b)) in AMPHTML,  because publishing this package outside of AMP's regular `gulp dist` won't pre-process them. So when trying to import these files, they won't necessarily be there.

I believe @kristoferbaxter has more experience publishing on npm and could provide more details.

**Comment by rsimha** [comment]
> because publishing this package outside of AMP's regular `gulp dist` won't pre-process them

_[Infra triage]_ Pinging this issue to ask for more clarity on the plans you're referring to re: publishing AMP to `npm`. 

**Some general context:** As of today, all of AMP's core runtime and extension JS binaries are built using `amp dist`. If someone wishes to use AMP without the CDN for some reason, they can use the open-sourced `amp release` workflow, which uses `amp dist` under the covers, to generate binaries.

We currently don't have plans of changing this mechanism for regular AMP. The only exception is the Bento effort, for which the npm publishing plans are still in review.
